### PR TITLE
feat(rule): update pick-up move rule [CARROT-1031]

### PR DESCRIPTION
**Pull request by RafaPalau**
Summary
This PR updates the business rule to verify metadata for events.

Details

Before: Verification if the metadata `move-type=pick-up` exists in the `OPEN` or `MOVE` event.
After: Verification if the metadata `move-type=pick-up` or `move-type=shipment-request` exists in the `OPEN` event.

### Related links

Task: https://app.clickup.com/t/3005225/CARROT-1031

---

- [X] **I, the PR author, declare that this PR works as expected and does not break any service. I also declare that I gave my best to apply all of the best practices and that I'm leaving the code better than I found it.**

## Summary by CodeRabbit

- **New Features**
  - Enhanced event processing to focus on OPEN events with specific attributes like PICK_UP and SHIPMENT_REQUEST.
  - Added a new scenario for OPEN events with SHIPMENT_REQUEST attribute.

- **Improvements**
  - Updated descriptions for event attributes and scenarios to provide more precise information.

- **Bug Fixes**
  - Adjusted logic to correctly handle event processing for PICK_UP and SHIPMENT_REQUEST attributes.

### CODE REVIEW

File: apps/methodologies/bold/rule-processors/mass/pick-up-move/src/lib/pick-up-move.processor.spec.ts [hunk 1]
```diff
@@ -38,13 +38,34 @@ describe('PickUpMoveProcessor', () => {
                 },
               ],
             },
-            name: random<DocumentEventName.MOVE | DocumentEventName.OPEN>(),
+            name: random<DocumentEventName.OPEN>(),
```
Comment (cris-santos) [inline apps/methodologies/bold/rule-processors/mass/pick-up-move/src/lib/pick-up-move.processor.spec.ts]: nesse caso, poderia remover o random e passar o evento de OPEN direto

File: apps/methodologies/bold/rule-processors/mass/pick-up-move/src/lib/pick-up-move.processor.spec.ts [hunk 2]
```diff
@@ -38,13 +38,34 @@ describe('PickUpMoveProcessor', () => {
...
+            metadata: {
+              attributes: [
+                {
+                  isPublic: true,
+                  name: DocumentEventAttributeName.MOVE_TYPE,
+                  value: DocumentEventMoveType.SHIPMENT_REQUEST,
+                },
+              ],
+            },
+            name: random<DocumentEventName.OPEN>(),
```
Comment (cris-santos) [inline apps/methodologies/bold/rule-processors/mass/pick-up-move/src/lib/pick-up-move.processor.spec.ts]: nesse caso, poderia remover o random e passar o evento de OPEN direto

**Review by cris-santos — DISMISSED** [review]

File: apps/methodologies/bold/rule-processors/mass/pick-up-move/src/lib/pick-up-move.processor.spec.ts (hunk 2 above)
Comment (RafaPalau) [reply apps/methodologies/bold/rule-processors/mass/pick-up-move/src/lib/pick-up-move.processor.spec.ts]: Removido [aqui](https://github.com/carrot-foundation/audit-rules/pull/36/commits/29a2c7cede6cd9e85a15e07e06ccc6cf3c5cb6c3)

File: apps/methodologies/bold/rule-processors/mass/pick-up-move/src/lib/pick-up-move.processor.spec.ts (hunk 1 above)
Comment (RafaPalau) [reply apps/methodologies/bold/rule-processors/mass/pick-up-move/src/lib/pick-up-move.processor.spec.ts]: Removido [aqui](https://github.com/carrot-foundation/audit-rules/pull/36/commits/29a2c7cede6cd9e85a15e07e06ccc6cf3c5cb6c3)

**Review by AMarcosCastelo — APPROVED** [review]

**Pull request merged**